cumulus_etl/formats/deltalake.py

```diff
@@ -161,52 +162,11 @@ def _configure_fs(root: store.Root, spark: pyspark.sql.SparkSession):
             spark.conf.set("fs.s3a.endpoint.region", region_name)
 
     @contextlib.contextmanager
-    def pandas_to_spark_with_schema(self, dataframe: pandas.DataFrame) -> pyspark.sql.DataFrame | None:
-        """Transforms a pandas DF to a spark DF with a full FHIR schema included"""
-        # This method solves two problems:
-        # 1. Pandas schemas are very loosey-goosey (and don't really take nested data into account), so simply
-        #    calling self.spark.createDataFrame(df) does not give names for nested struct fields.
-        # 2. We want to provide column info for all valid FHIR fields (at least shallowly) so that the
-        #    downstream SQL can reference all toplevel columns even if the source data doesn't have those fields.
-        #
-        # Issue #1 is solved by writing to parquet and reading it back in (a little wonky, but it gives full schemas).
-        # Issue #2 is solved by merging a computed FHIR schema with the actual data schema.
-        #
-        # Some devils-in-the-details:
-        # - Pyspark does not let us merge schemas in python code, we can only seem to do it while reading in multiple
-        #   dataframes. So we write out the schema as an empty parquet file and read it back in with the data.
-        #   We could write some manual schema-merging code, but I'm leery that we'd get it right or that it's worth
-        #   doing ourselves rather than just writing this weird file and letting pyspark do it for us.
-        # - Our FHIR schema is incomplete and shallow (it skips all nested structs) to avoid infinite recursion issues,
-        #   and we simply merge this incomplete schema in with the actual data schema, which will have full nested
-        #   inferred schemas for exactly the fields it uses. We always write to the delta lake with autoMerge of
-        #   schemas enabled, so incrementally adding fields to existing lakes will be fine.
-        # - Delta Lake does not like columns that have null types nor struct types with no children. So we make sure
-        #   that every column has *some* definition, and that structs have content.
-
+    def pyarrow_to_spark(self, table: pyarrow.Table) -> pyspark.sql.DataFrame | None:
+        """Transforms a pyarrow table to a spark DF"""
+        # There must be a better way?
```

Comment:
oh dear, one only hopes.

Reply:
In the grand scheme of things, this is a small oddity - we were writing out pandas to parquet and reading back into sparks before (to get proper schemas by way of pyarrow). And now that we have pyarrow directly, I'm just swapping in pyarrow for pandas.

Searching for how to do direct pyarrow -> pyspark yields people talking about doing this (parquet intermediate) or going to pandas as an intermediate (but that way lies bad schemas, which is why even when we used pandas, we wrote to a parquet).

The native-rust [python bindings](https://github.com/delta-io/delta-rs/blob/main/python/deltalake/table.py) seem to be very pyarrow focused, so that's promising for the future...

Reply:
I added a fuller comment to this now that `deltalake.py` is taking in a list of dicts rather than a pyarrow Table. I know of a better way to do this, but I'm leaving it for a future improvement:

```
        # This is the quick and dirty way - write batch to parquet with pyarrow and read it back.
        # But a more direct way would be to convert the pyarrow schema to a pyspark schema and just
        # call self.spark.createDataFrame(batch.rows, schema=pyspark_schema). A future improvement.
```